core/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -140,6 +140,18 @@ public ConnectionManager(final AblyRealtime ably, Connection connection) {
 			Log.e(getClass().getName(), msg, e);
 			throw new RuntimeException(msg, e);
 		}
+
+		try {
+			connectivityStateChangeEmitter = (ConnectivityStateChangeEmitter) Class.forName("io.ably.lib.transport.ConnectivityStateChangeEmitterImpl").newInstance();
+			connectivityStateChangeEmitter.on(this);
+		} catch (ClassNotFoundException e) {
+			e.printStackTrace();
```

Comment:
> Should this not go to the logger?

It's an error that formally must be handled, but it indicates a packaging error with the library rather than any error that can realistically occur in practice. So either we know for sure from our build/CI process that it will never happen, and we just have an empty catch block, or we call the logger. On Android you won't necessarily see `System.err` in logcat, so if you're relying on this message to realise there's a problem in development, it should go to the logger.